tests/test_cnes_sonarqube.py

```diff
@@ -0,0 +1,251 @@
+        assert cnes_quality_gates # not empty
+        assert cnes_quality_gates[0]['isDefault']
+
+    def test_check_qp(self):
+        """
+        As a SonarQube user, I want the SonarQube server to have the
+        CNES Quality Profiles available so that I can use them.
+        """
+        required_quality_profiles = (
+            "CNES_JAVA_A",
+            "CNES_JAVA_B",
+            "CNES_JAVA_C",
+            "CNES_JAVA_D",
+            "CNES_PYTHON_A",
+            "CNES_PYTHON_B",
+            "CNES_PYTHON_C",
+            "CNES_PYTHON_D",
+            "CNES_CPP_A",
+            "CNES_CPP_B",
+            "CNES_CPP_C",
+            "CNES_CPP_D",
+            "CNES_C_A",
+            "CNES_C_B",
+            "CNES_C_C",
+            "CNES_C_D",
+            "CNES_C_EMBEDDED_A",
+            "CNES_C_EMBEDDED_B",
+            "CNES_C_EMBEDDED_C",
+            "CNES_C_EMBEDDED_D"
+        )
+        quality_profiles = requests.get(f"{self.SONARQUBE_URL}/api/qualityprofiles/search").json()['profiles']
+        cnes_quality_profiles = [ qp['name'] for qp in quality_profiles if re.match(r'CNES_\w+_[ABCD]', qp['name']) ]
+        for profile in required_quality_profiles:
+            # Hint: if this test fails, one or more Quality Profiles may be missing, check the container logs
+            assert profile in cnes_quality_profiles
+
+    def test_eus_admin(self):
+        """
+        Evil User Story:
+        As a hacker, I want to use the default admin password ("admin")
+        to log in as admin.
+        """
+        status = requests.post(f"{self.SONARQUBE_URL}/api/authentication/login",
+                    auth=("admin", "admin"),
+                    data={"login": "admin", "password": "admin"}).status_code
+        # Hint: if this test fails, a hacker may have logged in (HTTP 200) or crash the server (HTTP 500+)
+        assert status == 401 # Unauthorized
+
+    def test_no_config_twice(self):
+        """
+        As a SonarQube user, I want the configuration of the server
+        not to be executed if the server has already been configured
+        so that the database is not populated more than once.
+        """
+        # On Linux, max_map_count must be at least 262144 to run this test
+        if platform.system() == 'Linux':
+            proc_stdout = subprocess.run("sysctl -a", shell=True, check=True, capture_output=True).stdout
+            max_map_count = int(re.findall('vm.max_map_count = [0-9]+', str(proc_stdout))[0].split(' = ')[1])
+            # Hint: if this test fails, run: sudo sysctl -w vm.max_map_count=262144
+            assert max_map_count >= 262144
+        docker_client = docker.from_env()
+        lequalsonarqube_container_name="lequalsonarqube-compose"
+        # Use the compose file with an external database
+        print("Starting the service (sonarqube and postgres)...")
+        subprocess.run("docker-compose up -d", shell=True, check=True, capture_output=True)
```

Comment:
Codacy found an issue: [subprocess call with shell=True seems safe, but may be changed in the future, consider rewriting without shell](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)